From 73d3fa8e7a1028c89b20c74f6444a76a722d8041 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heiko.becker@kde.org>
Date: Sun, 18 Apr 2021 11:47:30 +0200
Subject: [PATCH] Include <glib.h> outside of the extern block

This fixes the build with glib >= 2.68. It's caused by [1] and glib
upstream did not want to include a workaround because the fixes
downstream seem trivial [2][3].

[1] https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1715
[2] https://gitlab.gnome.org/GNOME/glib/-/issues/2331
[3] https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1935
---
 src/migration/mdb/3rdparty/mdbtools/include/mdbtools.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/migration/mdb/3rdparty/mdbtools/include/mdbtools.h b/src/migration/mdb/3rdparty/mdbtools/include/mdbtools.h
index 31f48c171..56276e2d6 100644
--- a/src/migration/mdb/3rdparty/mdbtools/include/mdbtools.h
+++ b/src/migration/mdb/3rdparty/mdbtools/include/mdbtools.h
@@ -18,6 +18,8 @@
 #ifndef _mdbtools_h_
 #define _mdbtools_h_
 
+#include <glib.h>
+
 #ifdef __cplusplus
   extern "C" {
 #endif
@@ -32,7 +34,6 @@
 #endif
 #include <ctype.h>
 #include <string.h>
-#include <glib.h>
 
 #ifdef HAVE_ICONV
 #include <iconv.h>
-- 
GitLab

